### PR TITLE
feat(fetch): add common media content types enum

### PR DESCRIPTION
**Pull request by kaname-png**

### CODE REVIEW

File: packages/fetch/src/lib/types.ts [hunk 1]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	/**
+	 * The `TEXT/HTML` body media content type.
+	 * @see [Media-Types / TEXT/HTML](https://www.iana.org/assignments/media-types/text/html)
+	 */
+	TEXT_HTML = 'text/html',
+	/**
+	 * The `TEXT/PLAIN` body media content type.
+	 */
+	TEXT_PLAIN = 'text/plain'
+}
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:117]:
```suggestion
/**
 * A list of common [Media Content Types](https://www.iana.org/assignments/media-types/media-types.xhtml) as defined by the IANA.
 */
export const enum FetchMediaContentTypes {
	/**
	 * The `JSON` media content type.
	 * @see [Media-Types / JSON](https://www.iana.org/assignments/media-types/application/json)
	 */
	JSON = 'application/json',
	/**
	 * The `XML` media content type.
	 * @see [Media-Types / XML](https://www.iana.org/assignments/media-types/application/xml)
	 */
	XML = 'application/xml',
	/**
	 * The `X-WWW-FORM-URLENCODED` media content type.
	 * @see [Media-Types / X-WWW-FORM-URLENCODED](https://www.iana.org/assignments/media-types/application/x-www-form-urlencoded)
	 */
	FORM_URLENCODED = 'application/x-www-form-urlencoded',
	/**
	 * The `FORM-DATA` media content type.
	 * @see [Media-Types / FORM-DATA](https://www.iana.org/assignments/media-types/application/form-data)
	 */
	FORM_DATA = 'multipart/form-data',
	/**
	 * The `TEXT/HTML` media content type.
	 * @see [Media-Types / TEXT/HTML](https://www.iana.org/assignments/media-types/text/html)
	 */
	TEXT_HTML = 'text/html',
	/**
	 * The `TEXT/PLAIN` media content type.
	 */
	TEXT_PLAIN = 'text/plain'
}
```

They are just media content types, not Body media content types. Just read the IANA link you linked, it doesn't mention body at all:
> Procedures for registering Media Types can be found in [[RFC6838](https://www.iana.org/go/rfc6838)], [[RFC4289](https://www.iana.org/go/rfc4289)], 
and [[RFC6657](https://www.iana.org/go/rfc6657)]. Additional procedures for registering media types for transfer via 
Real-time Transport Protocol (RTP) can be found in [[RFC4855](https://www.iana.org/go/rfc4855)].

**Review by favna — CHANGES_REQUESTED** [review]

File: packages/fetch/src/lib/types.ts (hunk 1 above)
Comment (kyranet) [reply packages/fetch/src/lib/types.ts:117]: Why are all members in SCREAMING_CASE, when we exclusively use PascalCase for enums (and so does djs)?

**Review by kyranet — CHANGES_REQUESTED** [review]

File: packages/fetch/src/lib/types.ts (hunk 1 above)
Comment (kaname-png) [reply packages/fetch/src/lib/types.ts]: Yeah, I thought the same thing, I wasn't sure what to put

File: packages/fetch/src/lib/types.ts [hunk 2]
```diff
@@ -89,29 +89,29 @@ export const enum FetchMediaContentTypes {
...
 	 * @see [Media-Types / XML](https://www.iana.org/assignments/media-types/application/xml)
 	 */
-	XML = 'application/xml',
+	Xml = 'application/xml',
 	/**
 	 * The `X-WWW-FORM-URLENCODED` media content type.
 	 * @see [Media-Types / X-WWW-FORM-URLENCODED](https://www.iana.org/assignments/media-types/application/x-www-form-urlencoded)
 	 */
-	FORM_URLENCODED = 'application/x-www-form-urlencoded',
+	FormUrlEncoded = 'application/x-www-form-urlencoded',
```
Comment (pinkcig) [inline packages/fetch/src/lib/types.ts:102]: URL should be in upper-case

File: packages/fetch/src/lib/types.ts [hunk 3]
```diff
@@ -89,29 +89,29 @@ export const enum FetchMediaContentTypes {
...
 	 * @see [Media-Types / FORM-DATA](https://www.iana.org/assignments/media-types/application/form-data)
 	 */
-	FORM_DATA = 'multipart/form-data',
+	FormData = 'multipart/form-data',
 	/**
 	 * The `TEXT/HTML` media content type.
 	 * @see [Media-Types / TEXT/HTML](https://www.iana.org/assignments/media-types/text/html)
 	 */
-	TEXT_HTML = 'text/html',
+	TextHtml = 'text/html',
```
Comment (pinkcig) [inline packages/fetch/src/lib/types.ts:112]: Same as above

File: packages/fetch/src/lib/types.ts [hunk 4]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	/**
+	 * The `JSON` media content type.
+	 * @see [Media-Types / JSON](https://www.iana.org/assignments/media-types/application/json)
+	 */
+	Json = 'application/json',
+	/**
+	 * The `XML` media content type.
+	 * @see [Media-Types / XML](https://www.iana.org/assignments/media-types/application/xml)
+	 */
+	Xml = 'application/xml',
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:97]:
```suggestion
	XML = 'application/xml',
```

File: packages/fetch/src/lib/types.ts [hunk 5]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+
+/**
+ * A list of common [Media Content Types](https://www.iana.org/assignments/media-types/media-types.xhtml) as defined by the IANA.
+ */
+export const enum FetchMediaContentTypes {
+	/**
+	 * The `JSON` media content type.
+	 * @see [Media-Types / JSON](https://www.iana.org/assignments/media-types/application/json)
+	 */
+	Json = 'application/json',
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:92]:
```suggestion
	JSON = 'application/json',
```

File: packages/fetch/src/lib/types.ts [hunk 6]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	/**
+	 * The `XML` media content type.
+	 * @see [Media-Types / XML](https://www.iana.org/assignments/media-types/application/xml)
+	 */
+	Xml = 'application/xml',
+	/**
+	 * The `X-WWW-FORM-URLENCODED` media content type.
+	 * @see [Media-Types / X-WWW-FORM-URLENCODED](https://www.iana.org/assignments/media-types/application/x-www-form-urlencoded)
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:100]:
```suggestion
	 * @see [Media-Types / application / x-www-form-urlencoded](https://www.iana.org/assignments/media-types/application/x-www-form-urlencoded)
```

File: packages/fetch/src/lib/types.ts [hunk 7]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	 */
+	FormUrlEncoded = 'application/x-www-form-urlencoded',
+	/**
+	 * The `FORM-DATA` media content type.
+	 * @see [Media-Types / FORM-DATA](https://www.iana.org/assignments/media-types/application/form-data)
+	 */
+	FormData = 'multipart/form-data',
+	/**
+	 * The `TEXT/HTML` media content type.
+	 * @see [Media-Types / TEXT/HTML](https://www.iana.org/assignments/media-types/text/html)
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:110]:
```suggestion
	 * @see [Media-Types / text / html](https://www.iana.org/assignments/media-types/text/html)
```

File: packages/fetch/src/lib/types.ts [hunk 8]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	 */
+	Xml = 'application/xml',
+	/**
+	 * The `X-WWW-FORM-URLENCODED` media content type.
+	 * @see [Media-Types / X-WWW-FORM-URLENCODED](https://www.iana.org/assignments/media-types/application/x-www-form-urlencoded)
+	 */
+	FormUrlEncoded = 'application/x-www-form-urlencoded',
+	/**
+	 * The `FORM-DATA` media content type.
+	 * @see [Media-Types / FORM-DATA](https://www.iana.org/assignments/media-types/application/form-data)
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:105]:
```suggestion
	 * @see [Media-Types / multipart / form-data](https://www.iana.org/assignments/media-types/multipart/form-data)
```

File: packages/fetch/src/lib/types.ts [hunk 9]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	 */
+	FormData = 'multipart/form-data',
+	/**
+	 * The `TEXT/HTML` media content type.
+	 * @see [Media-Types / TEXT/HTML](https://www.iana.org/assignments/media-types/text/html)
+	 */
+	TextHtml = 'text/html',
+	/**
+	 * The `TEXT/PLAIN` media content type.
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:114]:
```suggestion
	 * The `text/plain` media content type.
```

File: packages/fetch/src/lib/types.ts [hunk 10]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	 * @see [Media-Types / X-WWW-FORM-URLENCODED](https://www.iana.org/assignments/media-types/application/x-www-form-urlencoded)
+	 */
+	FormUrlEncoded = 'application/x-www-form-urlencoded',
+	/**
+	 * The `FORM-DATA` media content type.
+	 * @see [Media-Types / FORM-DATA](https://www.iana.org/assignments/media-types/application/form-data)
+	 */
+	FormData = 'multipart/form-data',
+	/**
+	 * The `TEXT/HTML` media content type.
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:109]:
```suggestion
	 * The `text/html` media content type.
```

File: packages/fetch/src/lib/types.ts [hunk 11]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	 * @see [Media-Types / XML](https://www.iana.org/assignments/media-types/application/xml)
+	 */
+	Xml = 'application/xml',
+	/**
+	 * The `X-WWW-FORM-URLENCODED` media content type.
+	 * @see [Media-Types / X-WWW-FORM-URLENCODED](https://www.iana.org/assignments/media-types/application/x-www-form-urlencoded)
+	 */
+	FormUrlEncoded = 'application/x-www-form-urlencoded',
+	/**
+	 * The `FORM-DATA` media content type.
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:104]:
```suggestion
	 * The `multipart / form-data` media content type.
```

File: packages/fetch/src/lib/types.ts [hunk 12]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	 * @see [Media-Types / JSON](https://www.iana.org/assignments/media-types/application/json)
+	 */
+	Json = 'application/json',
+	/**
+	 * The `XML` media content type.
+	 * @see [Media-Types / XML](https://www.iana.org/assignments/media-types/application/xml)
+	 */
+	Xml = 'application/xml',
+	/**
+	 * The `X-WWW-FORM-URLENCODED` media content type.
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:99]:
```suggestion
	 * The `application/x-www-form-urlencoded` media content type.
```

File: packages/fetch/src/lib/types.ts [hunk 13]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+ * A list of common [Media Content Types](https://www.iana.org/assignments/media-types/media-types.xhtml) as defined by the IANA.
+ */
+export const enum FetchMediaContentTypes {
+	/**
+	 * The `JSON` media content type.
+	 * @see [Media-Types / JSON](https://www.iana.org/assignments/media-types/application/json)
+	 */
+	Json = 'application/json',
+	/**
+	 * The `XML` media content type.
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:94]:
```suggestion
	 * The `application/xml` media content type.
```

File: packages/fetch/src/lib/types.ts [hunk 14]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	 * @see [Media-Types / XML](https://www.iana.org/assignments/media-types/application/xml)
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:95]:
```suggestion
	 * @see [Media-Types / application / xml](https://www.iana.org/assignments/media-types/application/xml)
```

File: packages/fetch/src/lib/types.ts [hunk 15]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
 	 */
 	Patch = 'PATCH'
 }
+
+/**
+ * A list of common [Media Content Types](https://www.iana.org/assignments/media-types/media-types.xhtml) as defined by the IANA.
+ */
+export const enum FetchMediaContentTypes {
+	/**
+	 * The `JSON` media content type.
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:89]:
```suggestion
	 * The `application/json` media content type.
```

File: packages/fetch/src/lib/types.ts [hunk 16]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
...
+	 * @see [Media-Types / JSON](https://www.iana.org/assignments/media-types/application/json)
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:90]:
```suggestion
	 * @see [Media-Types / application / json](https://www.iana.org/assignments/media-types/application/json)
```

File: packages/fetch/src/lib/types.ts [hunk 17]
```diff
@@ -80,3 +80,38 @@ export const enum FetchMethods {
 	 */
 	Patch = 'PATCH'
 }
+
+/**
+ * A list of common [Media Content Types](https://www.iana.org/assignments/media-types/media-types.xhtml) as defined by the IANA.
```
Comment (favna) [inline packages/fetch/src/lib/types.ts:85]:
```suggestion
 * A list of common [Media Content Types](https://www.iana.org/assignments/media-types/media-types.xhtml) as defined by the [IANA](https://www.iana.org/).
 * Media Content Types are also known as a Multipurpose Internet Mail Extensions or MIME type
 * Media Content Types are defined and standardized in IETF's [RFC 6838](https://datatracker.ietf.org/doc/html/rfc6838). 
```

**Review by favna — CHANGES_REQUESTED** [review]
Also add the following media types:
- `application/javascript`
- [`text/javascript`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#textjavascript)
- [`text/css`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#textcss)
- [`application/octet-stream`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#applicationoctet-stream)
- [`image/gif`](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types#gif_graphics_interchange_format)
- [`image/jpeg`](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types#jpeg_joint_photographic_experts_group_image)
- [`image/png`](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types#png_portable_network_graphics)
- [`image/webp`](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types#webp_image)
- [`image/apng`](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types#apng_animated_portable_network_graphics)
- `audio/webm`
- `video/webm`
- `audio/ogg`
- `video/ogg`
- `audio/wav`

**Comment by kaname-png** [comment]
@favna Done.
Btw, the `text/javascript` content type is obsoleted.
image

**Comment by favna** [comment]
> Btw, the `text/javascript` content type is obsoleted.

Oh my bad. Good that you didn't include it. Anyway I perused the IANA list a bit more and found a bunch more to include. Rather than bombarding with review comments I added them myself. Now if only you would be so kind as to alphabetically sort all the types that are there right now, that would be grand.

**Comment by kaname-png** [comment]
hmm, but then this pr stopped being common content types, right?

**Comment by favna** [comment]
> hmm, but then this pr stopped being common content types, right?

Oh don't worry, this is still far, faaaarrr from all of them. I grabbed only the common audio, video, and font formats.

**Comment by favna** [comment]
Just 1 note @kaname-png, once you make the alphabetically sorted change the unit test I just added will probably fail. Just run a sort alphabetically in vscode for the second expect and it'll pass I think.

**Review by favna — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**